src/Symfony/Component/Process/Process.php

```diff
@@ -297,15 +304,15 @@ public function start($callback = null)
 
         $this->process = proc_open($commandline, $descriptors, $this->processPipes->pipes, $this->cwd, $this->env, $this->options);
 
-        if ($ptsWorkaround) {
-            fclose($ptsWorkaround);
-        }
-
         if (!is_resource($this->process)) {
             throw new RuntimeException('Unable to launch a new process.');
         }
         $this->status = self::STATUS_STARTED;
 
+        if (isset($descriptors[3])) {
+            fclose($ptsWorkaround);
```

Comment:
>
> I don't think so. they are closed at the end of the process
>
I tried with this script:
```php
for ($i = 100000; $i; --$i) {
    $h[] = fopen(__FILE__, 'r');
}
```

if fails as expected with `failed to open stream: Too many open files`

but if I replace `$h[]` by `$h` it doesn't fail, which for me means the
files are closed once resources are freed. Isn't it?